java/jar/ext.go

```diff
@@ -0,0 +1,17 @@
+package jar
+
+import "path"
+
+// ValidExt determines if the file name extension
+// is a valid JAR file extension.
+func ValidExt(name string) bool {
+	ext := path.Ext(name)
```

Comment:
I have seen a few instances of `path` used and other instances of `filepath` being used. How do you usually determine when to use which? I tend to exclusively use `path` for URL related stuff. I believe it also makes sense for `zip`. However, the original `checkExt` used `filepath` and another check used `path`, and I can't determine why

Reply:
IFAIK filepath should be used when dealing with files at the `os` level as it has the ability to make determinations based on the os filesystem and path should be used when the dealing with generic related paths.

Looks like for `Ext()` specifically the functions are the same save for some os compatibility guarding in `filepath.Ext`. I'd say you're right in using path as the os (whatever Clair is running on) has little to do with what is being inspected.